common/inventory_parser/inventory_parser.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/inventory_parser.hpp>
+
+#include <boost/filesystem.hpp>
+#include <common/key_value_parser.hpp>
+#include <common/processes.hpp>
+#include <common/log.hpp>
+
+namespace mender {
+namespace common {
+namespace inventory_parser {
+
+using namespace std;
+namespace kvp = mender::common::key_value_parser;
+namespace procs = mender::common::processes;
+namespace log = mender::common::log;
+namespace err = mender::common::error;
+namespace fs = boost::filesystem;
+
+kvp::ExpectedKeyValuesMap GetInventoryData(const string generators_dir) {
+	bool any_success;
```

Comment:
S: Initialize them also

Reply:
Isn't `any_success` pretty much the same as checking if the map has any values?
Which also begs the question, is this necessarily an error? :thinking:

Reply:
> Isn't any_success pretty much the same as checking if the map has any values?

It is, but I like the explicit name and explicit assignment.

>  Which also begs the question, is this necessarily an error?

Hard to say, what exactly should be treated as error. But to me it makes sense to treat no scripts as not an error, some successful and some failing scripts as not an error with issues being logged and all scripts failing as an error. Then again, not sure what a general approach to things like this is/should be in mender.

Reply:
Yeh, makes sense :+1: 

I would still initialize the boolean I think though

Reply:
> I would still initialize the boolean I think though